addons/project/views/project_views.xml

```diff
@@ -387,7 +390,7 @@
                             name="%(act_project_project_2_project_task_all)d" icon="fa-tasks">
                             <field string="Tasks In Progress" name="task_count" widget="statinfo" options="{'label_field': 'label_tasks'}"/>
                         </button>
-                        <button class="oe_stat_button" name="action_get_list_view" type="object" icon="fa-check-square-o">
+                        <button class="oe_stat_button" name="action_get_list_view" type="object" icon="fa-check-square-o" attrs="{'invisible': [('allow_milestones', '=', False)]}" groups="project.group_project_milestone">
```

Comment:
```suggestion
                        <button class="oe_stat_button" name="action_get_list_view" type="object" icon="fa-check-square-o" attrs="{'invisible': [('allow_milestones', '=', False)]}">
```
I think the group is unnecessary here as it is already taken into account when computing `allow_milestones`